From stable-bounces@linux.kernel.org Wed Mar  7 18:52:15 2007
From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Wed, 07 Mar 2007 18:50:54 -0800 (PST)
Subject: Fix UDP header pointer after pskb_trim_rcsum()
To: stable@kernel.org
Cc: bunk@stusta.de
Message-ID: <20070307.185054.59656687.davem@davemloft.net>


From: Herbert Xu <herbert@gondor.apana.org.au>

[UDP]: Reread uh pointer after pskb_trim

The header may have moved when trimming.

Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 net/ipv4/udp.c |    1 +
 1 file changed, 1 insertion(+)

--- a/net/ipv4/udp.c
+++ b/net/ipv4/udp.c
@@ -1214,6 +1214,7 @@ int __udp4_lib_rcv(struct sk_buff *skb, 
 
 		if (ulen < sizeof(*uh) || pskb_trim_rcsum(skb, ulen))
 			goto short_packet;
+		uh = skb->h.uh;
 
 		udp4_csum_init(skb, uh);
 
